components/PropsTable.tsx

```diff
@@ -0,0 +1,124 @@
+import docgen from '@atom-learning/components/dist/docgen.json'
+import * as React from 'react'
+
+import { styled } from '../stitches.config'
+import { Box, InlineCode, Link, Text } from './'
+
+type PropsTableProps = {
+  for: React.FC
+}
+
+const columns = ['Prop', 'Type', 'Default value', 'Required']
+
+const Table = styled('table', {
+  borderCollapse: 'collapse',
+  width: '100%'
+})
+const Cell = styled('td', {
+  borderBottom: '1px solid $tonal300',
+  color: '$tonal800',
+  fontSize: '$md',
+  fontFamily: '"Inter"',
+  py: '$3',
+  pr: '$3',
+  textAlign: 'left',
+  verticalAlign: 'middle',
+  variants: {
+    appearance: {
+      heading: {
+        fontSize: '$sm',
+        whiteSpace: 'nowrap',
+        fontWeight: 600
+      }
+    }
+  }
+})
+
+const IconCheckmark = () => (
+  <svg
```

Comment:
this is a non-comment, once the icons are ready in the `components` library we should use that